api/proto/teleport/mfa/v1/mfa.proto

```diff
@@ -74,3 +77,20 @@ enum ChallengeAllowReuse {
   // Reuse is not permitted.
   CHALLENGE_ALLOW_REUSE_NO = 2;
 }
+
+// User verification requirement for the challenge.
+// See https://www.w3.org/TR/webauthn-2/#enum-userVerificationRequirement.
+enum UserVerificationRequirement {
+  // User verification requirement not specified.
+  // Functionally equivalent to VERIFICATION_REQUIREMENT_DISCOURAGED.
+  USER_VERIFICATION_REQUIREMENT_UNSPECIFIED = 0;
+  // User verification required.
+  // The server will reject challenge responses with UV=0.
+  USER_VERIFICATION_REQUIREMENT_REQUIRED = 1;
+  // User verification is desirable but not mandatory.
+  // The server will accept challenge responses with any UV value.
+  USER_VERIFICATION_REQUIREMENT_PREFERRED = 2;
```

Comment:
I'm a bit torn about a few characteristics here, if y'all want to chime in:

1. This could be a plain string instead of an enum, using `protocol.UserVerificationRequirement` constants directly. That is the route I took on the [webauthn protos][1] because, well, we don't really own this enum - the WebAuthn spec does.

2. I'm also a bit torn about exposing PREFERRED, as it's often an uninteresting setting. Arguably we only ever need REQUIRED, but I added all for completeness. (I suppose being able to exclude values is a bonus for the enum.)


[1]: https://github.com/gravitational/teleport/blob/b60e7a7a1f9f7b7818a5fe6ce1e8f0c1e2642023/api/proto/teleport/legacy/types/webauthn/webauthn.proto#L75

Reply:
Changed to string. Thanks for chiming in.